provisioning/baremetal_webhook.go

```diff
@@ -0,0 +1,159 @@
+package provisioning
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/pkg/errors"
+	admissionregistration "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	osclientset "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+)
+
+const (
+	validatingWebhookService           = "baremetal-operator-webhook-service"
```

Comment:
One day we'll finally resurrect the metrics PR (#99) and there'll be another thing that needs to point to the baremetal-operator. I wonder if it's worth naming this in such a way that it is not webhook-specific.

Reply:
If I understand correct, you mean once metrics api is ready, we can use this service with additional port bindings. If we rename this to general purpose, maybe we should move this service from `baremetal_webhook.go` to `baremetal_service.go`.

IMO, Instead re-using this service for different port bindings, I'd rather propose we can create new service resource (with same labels) in `baremetal_metrics.go` or even better metrics service yaml under manifests.

Reply:
Yeah, the current PR creates a separate Service for metrics.
I'm just wondering how many Services we need pointing to the same Container in the same Pod.